core/src/main/scala/org/bitcoins/core/psbt/PSBTHelper.scala

```diff
@@ -0,0 +1,269 @@
+package org.bitcoins.core.psbt
+
+import org.bitcoins.core.crypto.{ECDigitalSignature, ECPublicKey, ExtPublicKey}
+import org.bitcoins.core.hd.BIP32Path
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.CompactSizeUInt
+import org.bitcoins.core.protocol.script.{ScriptPubKey, ScriptSignature}
+import org.bitcoins.core.protocol.transaction.{Transaction, TransactionOutput}
+import org.bitcoins.core.psbt.GlobalPSBTRecord._
+import org.bitcoins.core.psbt.InputPSBTRecord._
+import org.bitcoins.core.psbt.PSBTGlobalKey._
+import org.bitcoins.core.psbt.PSBTInputKey._
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.core.serializers.script.RawScriptWitnessParser
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+private[psbt] case class PSBTParseResult[T <: PSBTMap](
+    remainingBytes: ByteVector,
+    maps: Vector[T])
+
+object PSBTHelper {
```

Comment:
Also, I thought I had left this comment on previous review but I couldn't find the comment so assuming it got deleted before I submitted it somehow. I would prefer that the `PSBTHelper` object be broken up into companion objects for things in `PSBT.scala`, if you want to keep it in a separate file I'm okay with that, maybe name it `PSBTCompanionObjects.scala` or `PSBTUtils.scala`. But what I want is to have a `PSBTRecord` object, as well as a `GlobalMap`, `InputMap`, and `OutputMap` object so that rather than calling `PSBTHelper.parseKeyAndValue`, I would instead call `PSBTRecord.fromBytes` and instead of `PSBTHelper.parseGlobalMap` I'd have `GlobalMap.fromBytes` and so on with the other two objects